account-kit/rn-signer/src/signer.ts

```diff
@@ -0,0 +1,46 @@
+import { z } from "zod";
+import {
+  BaseAlchemySigner,
+  BaseSignerClient,
+  DEFAULT_SESSION_MS,
+} from "@account-kit/signer";
+// eslint-disable-next-line import/extensions
+import { RNSignerClient, type RNSignerClientParams } from "./client";
+
+const SessionManagerParamsSchema = z.object({
+  sessionKey: z.string().default("alchemy-signer-session"),
+  storage: z
+    .enum(["localStorage", "sessionStorage"])
+    .default("localStorage") // We use a localStorage polyfill for React Native
+    .or(z.custom<Storage>()),
+  expirationTimeMs: z
+    .number()
+    .default(DEFAULT_SESSION_MS)
+    .describe(
+      "The time in milliseconds that a session should last before expiring [default: 15 minutes]"
+    ),
+  client: z.custom<BaseSignerClient>(),
+});
+
+const RNAlchemySignerParamsSchema = z
+  .object({
+    client: z.custom<RNSignerClientParams>(),
+  })
+  .extend({
+    sessionConfig: SessionManagerParamsSchema.omit({
+      client: true,
+    }).optional(),
```

Comment:
I would just import from signer as above and then do this here:

```suggestion
    sessionConfig: SessionManagerParamsSchema.omit({
      client: true, storage: true
    }).optional(),
```